### PR TITLE
[image_picker] Update to 1.0

### PR DESCRIPTION
**Pull request by stuartmorgan**
The plugin now covers all of our supported platforms, most highly-requested features have been addressed, and the API has been stable for quite some time (in that no breaking changes have been required). Given that, a 1.0 version number reflects the current state of the package than a 0.x version.

As part of the breaking change, the `get*` methods that were deprecated several years ago have now been removed.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/image_picker/image_picker/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
+  `XFile` rather than `PickedFile` for return values.
+    * As this is the only change, we encourage authors of published packages
+      that depend on `image_picker` to consider using a constraint of
+      `'>=0.8.9 <2.0.0'` rather than `^1.0.0` when updating dependencies, to
+      avoid conflicts with packages that have not yet updated.
+
 ## 0.8.9
 
-* Adds `getMedia` and `getMultipleMedia` methods.
+* Adds `pickMedia` and `pickMultipleMedia` methods.
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker/CHANGELOG.md:13]: While I generally don't want us changing old changelogs, I noticed that we forgot to adjust the names for the app-facing package, making this wrong, and it's actively confusing when combined with the discussion of removing `get` in 1.0.0 so I thought it was worthwhile here.

Comment (tarrinneal) [reply packages/image_picker/image_picker/CHANGELOG.md:13]: good catch

File: packages/image_picker/image_picker/README.md [hunk 2]
```diff
@@ -186,15 +186,13 @@ final XFile? media = await picker.pickMedia();
 final List<XFile> medias = await picker.pickMultipleMedia();
 ```
 
-## Migrating to 0.8.2+
+## Migrating to 1.0
 
-Starting with version **0.8.2** of the image_picker plugin, new methods have
-been added for picking files that return `XFile` instances (from the
+Starting with version 0.8.2 of the image_picker plugin, new methods were
+been added that return `XFile` instances (from the
```
Comment (tarrinneal) [inline packages/image_picker/image_picker/README.md:192]: -been

**Review by tarrinneal — APPROVED** [review]
Very exciting!

File: packages/image_picker/image_picker/README.md (hunk 2 above)
Comment (stuartmorgan) [reply packages/image_picker/image_picker/README.md]: Oops! Fixed.

**Pull request merged**